File: app/src/main/java/com/don/myapplication/java/Trimmable.java
package com.don.myapplication.java;

import java.lang.reflect.Field;

/**
 * Utility interface that trims all String fields of the implementing class.
 */
public interface Trimmable {

    default void trim(){
        for (Field field : this.getClass().getDeclaredFields()) {
            try {
                field.setAccessible(true);
                Object value = field.get(this);
                if (value != null){
                    if (value instanceof String){
                        String trimmed = (String) value;
                        field.set(this, trimmed.trim());
                    }
                }
            } catch(Exception e) {
                e.printStackTrace();
            }
        }
    }
}

File: app/src/main/java/com/don/myapplication/kotlin/Trimmable.kt
package com.don.myapplication.kotlin



/**
 * Utility interface that trims all String fields of the implementing class.
 */
interface Trimmable {

    /**
     * Trim all Strings
     */
    fun trim() {
        for (field in this.javaClass.declaredFields) {
            try {
                field.isAccessible = true
                val value = field.get(this)
                if (value != null) {
                    if (value is String) {
                        field.set(this, value.trim { it <= ' ' })
                    }
                }
            } catch (e: Exception) {
                e.printStackTrace()
            }

        }
    }
}